dotcom-rendering/src/components/YoutubeAtom/YoutubeAtom.tsx

```diff
@@ -134,16 +134,21 @@ export const YoutubeAtom = ({
 	 */
 	const showPlaceholder = (!hasOverlay || overlayClicked) && !playerReady;
 
-	let loadPlayer;
-	if (!hasOverlay) {
-		// load the player if there is no overlay
-		loadPlayer = true;
-	} else if (overlayClicked) {
-		// load the player if the overlay has been clicked
-		loadPlayer = true;
-	} else {
-		loadPlayer = false;
-	}
+	const [loadPlayer, setLoadPlayer] = useState<boolean>(false);
+	useEffect(() => {
+		if (elementId === 'server') {
+			// We may never load the player on the server
+			return;
+		}
+
+		if (!hasOverlay) {
+			// load the player if there is no overlay
+			setLoadPlayer(true);
+		} else if (overlayClicked) {
+			// load the player if the overlay has been clicked
+			setLoadPlayer(true);
+		}
+	}, [elementId, hasOverlay, overlayClicked]);
```

Comment:
I'm wondering if this could be simplified.

Instead of elementId could we pass through index: number | undefined and return in the main render path if isUndefined(index)?

I think that would mean the other state could remain derived and it would remove a useEffect and a re-render.

Reply:
Okay, I’ll keep the changes to a minimal and just add one more check:

```js
if (isUndefined(index)) {
	loadPlayer = false;
}
```